Marlin/src/gcode/calibrate/G33.cpp

```diff
@@ -409,17 +409,21 @@ void GcodeSuite::G33() {
   const bool probe_at_offset = TERN0(HAS_PROBE_XY_OFFSET, parser.boolval('O')),
                   towers_set = !parser.seen_test('T');
 
+  // needs to be moved to all delta config files
+  #define BED_DIAMETER 220
```

Comment:
Is there a way to do this in one go?

Reply:
The bed's diameter is assumed to be the `X_BED_SIZE` which should be equal to the `Y_BED_SIZE` on a delta. Is this `BED_DIAMETER` value meant to be smaller/larger than the specified bed size?

Reply:
On deltas : 
(Configuration.h)
```
#define X_BED_SIZE ((DELTA_PRINTABLE_RADIUS) * 2)
#define Y_BED_SIZE ((DELTA_PRINTABLE_RADIUS) * 2)

#define X_MIN_POS -(DELTA_PRINTABLE_RADIUS)
#define Y_MIN_POS -(DELTA_PRINTABLE_RADIUS)
#define X_MAX_POS DELTA_PRINTABLE_RADIUS
#define Y_MAX_POS DELTA_PRINTABLE_RADIUS
```
and
(Conditional_post.h)
```
#define X_MAX_LENGTH (X_MAX_POS - (X_MIN_POS))
#define Y_MAX_LENGTH (Y_MAX_POS - (Y_MIN_POS))
```
so `X_MAX_LENGTH` and `Y_MAX_LENGTH` are also set to `((DELTA_PRINTABLE_RADIUS) * 2)` and as such are always equal by definition to `X_BED_SIZE` and `Y_BED_SIZE`.

I need to know the size border on the bed that can't be used for printing but that can be used for (offsetted) probing at nozzle positions (blue area).

![radius](https://user-images.githubusercontent.com/26793057/133888540-6cd76454-7f97-4029-99b6-54b0316a2562.png)

Reply:
The old setting `DELTA_PROBEABLE_RADIUS` was replaced with the `DELTA_PRINTABLE_RADIUS - PROBING_MARGIN`, indicating the "region of the bed" which is able to be probed (because getting too close to the edge with an inductive or capacitive probe will produce bad readings).

Meanwhile, the functions `position_is_reachable` (`motion.h`) and `probe.can_reach` (`probe.h`) are used to check the physical limits on Delta motion. Since the probe offsets and the software endstops can both be changed at runtime, these functions are used instead of relying on some fixed value. The `position_is_reachable` function takes an extra `inset` parameter.

The reachable area outside of the bed on a delta might end up being required as a new setting, but it would be good to first consider whether the same thing can be accomplished with existing settings. And, it will be important to give some thought to `position_is_reachable` and `probe.can_reach` with any updates to delta settings.

Reply:
> Is there a way to do this in one go?

With some text editors (e.g., Sublime) you can do a global search-and-replace only within the `examples/delta` folder, searching for the line above (or below) the new addition, and replacing that line with: the existing line plus the new addition.